heartbeat/monitors/active/icmp/loop.go

```diff
@@ -92,13 +94,25 @@ func newICMPLoop() (*icmpLoop, error) {
 	// IPv4/IPv6 checking
 	conn4 := createListener("IPv4", "ip4:icmp")
 	conn6 := createListener("IPv6", "ip6:ipv6-icmp")
-
+	unprivilegedPossible := false
 	l := &icmpLoop{
 		conn4:    conn4,
 		conn6:    conn6,
 		recv:     make(chan packet, 16),
 		requests: map[requestID]*requestContext{},
 	}
+	if conn4 == nil && conn6 == nil {
+
+		switch runtime.GOOS {
+		case "darwin":
+		case "linux":
```

Comment:
Why linux only? Go has no fallthrough by default, but you can write `case "linux", "darwin":`.

What about other OSes (e.g. *BSD)? Would it make sense to create os specific init functions and use build tags (e.g. `open_unix.go`, `open_windows.go`) ?

Reply:
Indeed the switch case was wrong..  thanks!

As this is just a fall back plan, i don't see the need to split in two files. Default plan works on both systems, if and only if raw socket is not allowed, it will try to use unprivileged socket on linux and darwin. 
This was introduced in kernel in 2010 so in kernel years, it is too young..  it is not available on many distributions yet.